src/clr-angular/forms/common/layout.spec.ts

```diff
@@ -10,39 +10,213 @@ import { By } from '@angular/platform-browser';
 import { ClrLayout } from './layout';
 import { LayoutService, Layouts } from './providers/layout.service';
 
-@Component({ template: `<div clrForm clrLayout="vertical"></div>`, providers: [LayoutService] })
-class VerticalTest {}
-
-@Component({ template: `<div clrForm clrLayout="invalid"></div>`, providers: [LayoutService] })
-class InvalidTest {}
+function customTestComponentFactory(
+  layout: Layouts = Layouts.VERTICAL,
+  hasLabelSize: boolean = false,
+  labelSize: any = 2
+) {
+  @Component({
+    template: `<div clrForm clrLayout="${layout}" ${hasLabelSize ? `clrLabelSize="${labelSize}"` : ''}></div>`,
+    providers: [LayoutService],
+  })
+  class TestComponent {}
+  return TestComponent;
+}
 
 export default function(): void {
   describe('ClrLayout', () => {
     it('sets the layout selection in LayoutService on init if provided', function() {
-      TestBed.configureTestingModule({ declarations: [ClrLayout, VerticalTest] });
-      const fixture = TestBed.createComponent(VerticalTest);
+      const TestClass = customTestComponentFactory();
+      TestBed.configureTestingModule({ declarations: [ClrLayout, TestClass] });
+      const fixture = TestBed.createComponent(TestClass);
       const service = fixture.debugElement.injector.get(LayoutService);
       fixture.detectChanges();
+
       expect(service.layout).toEqual(Layouts.VERTICAL);
     });
 
     it('accepts layout option through layout input binding', function() {
-      TestBed.configureTestingModule({ declarations: [ClrLayout, VerticalTest] });
-      const fixture = TestBed.createComponent(VerticalTest);
+      const TestClass = customTestComponentFactory();
+      TestBed.configureTestingModule({ declarations: [ClrLayout, TestClass] });
+      const fixture = TestBed.createComponent(TestClass);
       const component = fixture.debugElement.query(By.directive(ClrLayout));
       const directive = component.injector.get(ClrLayout);
       const service = fixture.debugElement.injector.get(LayoutService);
       fixture.detectChanges();
-      expect(directive.layout).toEqual('vertical');
+
+      expect(directive.layout).toEqual(Layouts.VERTICAL);
       expect(service.layout).toEqual(Layouts.VERTICAL);
     });
 
     it('ignores invalid layout types', function() {
-      TestBed.configureTestingModule({ declarations: [ClrLayout, InvalidTest] });
-      const fixture = TestBed.createComponent(InvalidTest);
+      const TestClass = customTestComponentFactory(<Layouts>'invalid');
+      TestBed.configureTestingModule({ declarations: [ClrLayout, TestClass] });
+      const fixture = TestBed.createComponent(TestClass);
       const service = fixture.debugElement.injector.get(LayoutService);
       fixture.detectChanges();
+
       expect(service.layout).toEqual(Layouts.HORIZONTAL);
     });
+
+    it('horizontal layout label should have a default size of 2', () => {
```

Comment:
`it('label should have a default size of 2'), () => {`

Just test that the label is 2 by default.